torch/csrc/jit/script/module.h

```diff
@@ -222,53 +183,71 @@ struct TORCH_API Module {
   }
 
   std::shared_ptr<Module> get_module(const std::string& name) const {
-    return modules_[get_offset(name, EntityType::MODULE)];
+    auto obj = get_slot(name, EntityType::MODULE).value().toObject();
+    return std::make_shared<Module>(obj);
   }
 
-  c10::ArrayRef<std::shared_ptr<Module>> get_modules() const {
-    return modules_;
-  }
-  c10::ArrayRef<Slot> get_parameters() const {
-    return parameters_;
-  }
-  c10::ArrayRef<Slot> get_attributes() const {
-    return attributes_;
+  std::vector<std::shared_ptr<Module>> get_modules() const {
```

Comment:
I don't think is makes sense for these to be `shared_ptr` anymore, since Module is a value type anyway

Reply:
I agree, but this is a big breaking API change for anyone using this API. Any call like `foo.get_submodule("bar")->is_training()` will need to change because the thing being returned is no longer a pointer. We should do it but it will need to be in a follow up PR to minimize the amount of API changes this one causes.